lib/rubocop/config_loader_resolver.rb

```diff
@@ -239,8 +239,7 @@ def inherited_file(path, inherit_from, file)
     end
 
     def remote_file?(uri)
-      regex = URI::DEFAULT_PARSER.make_regexp(%w[http https])
-      /\A#{regex}\z/.match?(uri)
+      uri.start_with?('http://', 'https://')
```

Comment:
Seems like a simpler and sufficient solution, though I can do the `defined` check here as well.

Reply:
Might be good to check what prompted the more complex solution here in the first place. :-) I assume the simpler option will work for most people, but I want to be careful not to break someone's setup.

Reply:
I can do some spelunking in a bit. My initial assumption is that it would be broken anyways, since uri-like but invalid uris would fall through being loaded from disk. I haven't confirmed it though.

Reply:
Right, so initially this was a bit simpler with just `URI.regexp` but over time evolved into this because of Ruby changes and a bugfix https://github.com/rubocop/rubocop/commit/45a06913a56e4704a4e1269216d87613a6cb131e

It does indeed fall down to trying to load it from disk, which then fails. These files will never be valid because of file naming restrictions around `:` and `/`, so the only change will be a different error message if you do end up writing an invalid uri.

On that note, I improved the error message and added a test for it:
Previous:
> Configuration file not found: /home/user/code/rubocop/https:/invalid_üri/.rubocop.yml

New:
> Failed to resolve configuration: 'https://invalid_üri/.rubocop.yml' is not a valid URI

Because it now no longer falls through.